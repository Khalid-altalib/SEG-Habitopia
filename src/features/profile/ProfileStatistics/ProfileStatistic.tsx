import { Box, theme, useColorModeValue } from "native-base";
import React from "react";
import { Statistic, TextType } from "../../../../types";
import { LinearGradient } from "expo-linear-gradient";
import Text from "@components/Text";

type Props = {
  statistic: Statistic;
};

const ProfileStatistic = (props: Props) => {
  const { statistic } = props;
  const { name, quantity } = statistic;

  return (
    <Box overflow="hidden" rounded="lg">
      <LinearGradient
        colors={useColorModeValue(
          [theme.colors.white, theme.colors.gray[50]],
          [theme.colors.blueGray[600], theme.colors.blueGray[700]]
        )}
        start={[0, 0]}
        end={[1, 1]}
        style={{
          padding: 12.5,
          marginBottom: -9, // The -0 is to account for the text bottom margin/padding
        }}
      >
        <Text type={TextType.Small}>{name}</Text>
        <Text type={TextType.Heading}>{quantity}</Text>
      </LinearGradient>
    </Box>
  );
};

export default ProfileStatistic;
