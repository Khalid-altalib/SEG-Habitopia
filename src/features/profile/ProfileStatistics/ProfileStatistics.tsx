import { FlatList, Heading, View, VStack } from "native-base";
import React from "react";
import { Statistic } from "../../../../types";
import { useAppSelector } from "../../../app/hooks";
import ProfileStatistic from "./ProfileStatistic";

type Props = {};

const ProfileStatistics = (props: Props) => {
  const { error, loading, profile } = useAppSelector((state) => state.profile);
  //   const { statistics } = profile; // BACKEND_PLACEHOLDER

  const statistics: Statistic[] = [
    { name: "Streak", quantity: 5 },
    { name: "Wins", quantity: 1 },
    { name: "Check Ins", quantity: 42 },
    { name: "Level", quantity: 8 },
  ];

  return (
    <VStack>
      <Heading mb={4}>Statistics</Heading>
      <View flexWrap="wrap" flexDirection={"row"}>
        {statistics.map((statistic, i) => (
          <ProfileStatistic statistic={statistic} key={i} />
        ))}
      </View>
    </VStack>
  );
};

export default ProfileStatistics;
