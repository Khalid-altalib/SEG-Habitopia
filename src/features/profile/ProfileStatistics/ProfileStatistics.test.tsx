import React from "react";
import { render } from "@testing-library/react-native";
import ProfileStatistics from "./ProfileStatistics";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

jest.mock("@app/hooks", () => ({
  useSelector: jest.fn(() => ({
    profile: {
      statistics: [
        { name: "Followers", quantity: 100 },
        { name: "Following", quantity: 50 },
      ],
    },
  })),
}));

describe("ProfileStatistics", () => {
  const mockState = {
    profile: {
      profile: {
        userId: "123",
        name: "Test User",
        followerCount: 10,
        followingCount: 20,
        biography: "Test Biography",
      },
    },
  };

  const mockStore = configureStore([thunk])(mockState);
  let wrapper: any;
  beforeEach(() => {
    wrapper = render(
      <TestingWrapperNavigation store={mockStore}>
        <ProfileStatistics />
      </TestingWrapperNavigation>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
