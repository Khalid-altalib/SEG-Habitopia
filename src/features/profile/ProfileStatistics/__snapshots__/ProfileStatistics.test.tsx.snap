// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatistics renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 24,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 30,
                    "marginBottom": 16,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Statistics
              </Text>
              <View
                dataSet={{}}
                style={{}}
              />
              <View
                dataSet={{}}
                style={{}}
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "borderRadius": 8,
                      "overflow": "hidden",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    proxiedProperties={
                      {
                        "colors": [
                          4294967295,
                          4294638330,
                        ],
                        "endPoint": [
                          1,
                          1,
                        ],
                        "locations": undefined,
                        "startPoint": [
                          0,
                          0,
                        ],
                      }
                    }
                    style={
                      {
                        "marginBottom": -9,
                        "padding": 12.5,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Followers
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      100
                    </Text>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 12.5,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "borderRadius": 8,
                      "overflow": "hidden",
                    }
                  }
                >
                  <ViewManagerAdapter_ExpoLinearGradient
                    proxiedProperties={
                      {
                        "colors": [
                          4294967295,
                          4294638330,
                        ],
                        "endPoint": [
                          1,
                          1,
                        ],
                        "locations": undefined,
                        "startPoint": [
                          0,
                          0,
                        ],
                      }
                    }
                    style={
                      {
                        "marginBottom": -9,
                        "padding": 12.5,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      Following
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "color": "#18181b",
                          },
                          {
                            "fontFamily": undefined,
                            "fontSize": 40,
                          },
                          undefined,
                        ]
                      }
                    >
                      50
                    </Text>
                  </ViewManagerAdapter_ExpoLinearGradient>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
